RFS/src/main/java/com/rfs/common/SnapshotShardUnpacker.java

```diff
@@ -58,6 +72,36 @@ public void unpack(ShardMetadata.Data shardMetadata) {
         }
     }
 
+    @Override
+    public void close() {
+        try {
+            Path luceneIndexDir = Paths.get(luceneFilesBasePath + "/" + shardMetadata.getIndexName() + "/" + shardMetadata.getShardId());
```

Comment:
It looks like you're deleting a session directory here.  Why did you need to delete files above when tied to the stream?

Reply:
Not sure I understand the question.  Can you re-phrase?

Reply:
Is this redundant to what you are doing with the DeletingFileInputStream above?  This place feels like where all of your cached files should be deleted - or are there other files that are created outside of this class?

Reply:
Per discussion - there are two things that we need to clean up: the raw snapshot files we download from s3, and the files we convert those into which Lucene actually cares about.  This is for cleaning up the later.

Reply:
As per the [jira](https://opensearch.atlassian.net/browse/MIGRATIONS-1786) in the log message added above, we should eventually get this down to no extra file buffers, which reduces the number of resources that we're managing and should make the code more efficient and easier to maintain.